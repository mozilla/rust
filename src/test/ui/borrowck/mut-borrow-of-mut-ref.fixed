// run-rustfix
// Suggest not mutably borrowing a mutable reference
#![crate_type = "rlib"]

pub fn f(mut b: &mut i32) {
    h(&mut b);
    //~^ ERROR cannot borrow
}

pub fn g(mut b: &mut i32) {
    h(&mut &mut b);
    //~^ ERROR cannot borrow
}

pub fn h(_: &mut i32) {}
