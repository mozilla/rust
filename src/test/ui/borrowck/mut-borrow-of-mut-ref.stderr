error[E0596]: cannot borrow `b` as mutable, as it is not declared as mutable
  --> $DIR/mut-borrow-of-mut-ref.rs:6:7
   |
LL |     h(&mut b);
   |       ^^^^^^ cannot borrow as mutable
   |
help: consider making the binding mutable
   |
LL | pub fn f(mut b: &mut i32) {
   |          ^^^

error[E0596]: cannot borrow `b` as mutable, as it is not declared as mutable
  --> $DIR/mut-borrow-of-mut-ref.rs:11:12
   |
LL |     h(&mut &mut b);
   |            ^^^^^^ cannot borrow as mutable
   |
help: consider making the binding mutable
   |
LL | pub fn g(mut b: &mut i32) {
   |          ^^^

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0596`.
